duktape/src/main/jni/duktape/duk_config.h

```diff
@@ -3562,21 +3022,32 @@ typedef FILE duk_file;
 #if defined(DUK_USE_DATE_GET_NOW)
 /* External provider already defined. */
 #elif defined(DUK_USE_DATE_NOW_GETTIMEOFDAY)
-#define DUK_USE_DATE_GET_NOW(ctx)            duk_bi_date_get_now_gettimeofday((ctx))
+#define DUK_USE_DATE_GET_NOW(ctx)            duk_bi_date_get_now_gettimeofday()
 #elif defined(DUK_USE_DATE_NOW_TIME)
-#define DUK_USE_DATE_GET_NOW(ctx)            duk_bi_date_get_now_time((ctx))
+#define DUK_USE_DATE_GET_NOW(ctx)            duk_bi_date_get_now_time()
 #elif defined(DUK_USE_DATE_NOW_WINDOWS)
-#define DUK_USE_DATE_GET_NOW(ctx)            duk_bi_date_get_now_windows((ctx))
+#define DUK_USE_DATE_GET_NOW(ctx)            duk_bi_date_get_now_windows()
+#elif defined(DUK_USE_DATE_NOW_WINDOWS_SUBMS)
+#define DUK_USE_DATE_GET_NOW(ctx)            duk_bi_date_get_now_windows_subms()
 #else
 #error no provider for DUK_USE_DATE_GET_NOW()
 #endif
 
+/** Duktape-Android modification: Hack in date parsing and timezone support for Android */
+extern duk_int_t android__get_local_tzoffset(duk_double_t time);
+extern duk_bool_t android__date_parse_string(duk_context* ctx, const char* str);
+#define DUK_USE_DATE_GET_LOCAL_TZOFFSET android__get_local_tzoffset
+#define DUK_USE_DATE_PARSE_STRING android__date_parse_string
+/** End Duktape-Android. */
```

Comment:
This is unfortunate. Can we file an issue or something to allow easier external customization? Or is there info as to why that isn't allowed easily?

Reply:
Agreed.  I spent a bit of time trying to avoid this, but didn't want to hold the rest of the PR any longer.   I got to this point by looking at the example [here](https://github.com/svaarala/duktape/blob/master/examples/dummy-date-provider/dummy_date_provider.c).   I'll see if I can't find a better way (or request one) in a follow up.